src/parser/name_resolution.c

```diff
@@ -10530,13 +11279,285 @@ pt_resolve_dblink_server_name (PARSER_CONTEXT * parser, PT_NODE * node)
+      if (plkcol->col_list->type_enum == PT_TYPE_STAR && plkcol->col_list->info.name.resolved == NULL)
+	{
+	  return;		// case: * vs anything
+	}
+
+      if (resolved == NULL)
+	{			// col
+	  for (col = plkcol->col_list; col; col = col->next)
+	    {
+	      if (col->type_enum == PT_TYPE_STAR)
+		{
+		  return;	// case: tbl.* vs anything
+		}
+	      else if (intl_identifier_casecmp (col->info.name.original, original) == 0)
+		{
+		  return;	// case: col  vs col
+		}
+	    }
+	}
+      else if (original)
+	{			// tbl.col
+	  for (col = plkcol->col_list; col; col = col->next)
+	    {
+	      if (col->type_enum == PT_TYPE_STAR)
+		{
+		  return;	// case: tbl.* vs anything
+		}
+	      else if (intl_identifier_casecmp (col->info.name.original, original) == 0)
+		{
+		  return;	// case: col  vs col  or  col vs tbl.col
+		}
+	    }
+	}
+      else
+	{			// tbl.*
+	  if (plkcol->col_list->type_enum == PT_TYPE_STAR)
+	    {
+	      return;		// case: tbl.* vs tbl.*
+	    }
+
+	  parser_free_node (parser, plkcol->col_list);
+	  plkcol->col_list = NULL;
+	}
+    }
+
+  PT_NODE *name = parser_new_node (parser, PT_NAME);
+
+  if (resolved && original)
+    {
+      //name->info.name.resolved = pt_append_string (parser, NULL, resolved);
+      name->info.name.original = pt_append_string (parser, NULL, original);
+    }
+  else if (resolved)
+    {
+      name->type_enum = PT_TYPE_STAR;
+      name->info.name.resolved = pt_append_string (parser, NULL, resolved);
+    }
+  else
+    {
+      name->info.name.original = pt_append_string (parser, NULL, original);
+    }
+
+  plkcol->col_list = parser_append_node (name, plkcol->col_list);
+}
+
+
+static PT_NODE *
+pt_get_column_name_pre (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue_walk)
+{
+  S_LINK_COLUMNS *plkcol = (S_LINK_COLUMNS *) arg;
+  PT_NODE *name = NULL;
+
+  if (node->node_type == PT_SELECT)
+    {
+      *continue_walk = PT_STOP_WALK;
+    }
+  else if (node->node_type == PT_DOT_)
+    {				// case: tbl.col      
+      check_for_already_exists
+	(parser, plkcol, node->info.dot.arg1->info.name.original, node->info.dot.arg2->info.name.original);
+
+      *continue_walk = PT_LIST_WALK;
+    }
+  else if (node->node_type == PT_NAME)
+    {
+      if (node->type_enum == PT_TYPE_STAR)
+	{			// case:  tbl.*
+	  check_for_already_exists (parser, plkcol, node->info.name.original, NULL);
+	}
+      else
+	{
+	  check_for_already_exists (parser, plkcol, NULL, node->info.name.original);
+	}
+    }
+  else if (node->node_type == PT_VALUE && node->type_enum == PT_TYPE_STAR)
+    {
+      {
+	name = parser_new_node (parser, PT_NAME);
+	name->type_enum = PT_TYPE_STAR;
+	if (plkcol->col_list)
+	  {
+	    parser_free_node (parser, plkcol->col_list);
+	  }
+	plkcol->col_list = name;
+      }
+    }
+  return node;
+}
+
+static PT_NODE *
+pt_get_column_name_post (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue_walk)
+{
+  *continue_walk = PT_CONTINUE_WALK;
+  return node;
+}
```

Comment:
Why is this function required? Passing NULL to the post_function arg in `parser_walk_tree()` seems to be enough
```suggestion
(void) parser_walk_tree (parser, node_list,
			   pt_get_column_name_pre, plkcol, NULL, /*plkcol */ NULL);
```

Reply:
Ok, good point.